src/js/components/homepage/visualizations/TreeMap.jsx

```diff
@@ -0,0 +1,140 @@
+/**
+ * TreeMap.jsx
+ * Created by Emily Gullo 03/15/2017
+ **/
+
+import React from 'react';
+import * as d3 from 'd3';
+import _ from 'lodash';
+
+import Cell from './Cell';
+import * as Icons from '../../sharedComponents/icons/Icons';
+
+const propTypes = {
+    budgetCategories: React.PropTypes.object
+};
+
+export default class TreeMap extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            windowWidth: 0,
+            visualizationWidth: 0
+        };
+
+        this.handleWindowResize = _.throttle(this.handleWindowResize.bind(this), 50);
+    }
+
+    componentDidMount() {
+        this.handleWindowResize();
+        window.addEventListener('resize', this.handleWindowResize);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleWindowResize);
+    }
+
+    handleWindowResize() {
+        // determine if the width changed
+        const windowWidth = window.innerWidth;
+        if (this.state.windowWidth !== windowWidth) {
+            // width changed, update the visualization width
+            this.setState({
+                windowWidth,
+                visualizationWidth: this.sectionWrapper.offsetWidth
+            });
+        }
+    }
+
+    render() {
+        // put the data through d3's hierarchy system to sum and sort it
+        const root = d3.hierarchy(this.props.budgetCategories)
+        .sum((d) => (d.value))
+        .sort((a, b) => b.height - a.height || b.value - a.value);
+
+        // set up a treemap object and pass in the root
+        let tileStyle = d3.treemapBinary;
+        if (this.state.windowWidth < 768) {
+            tileStyle = d3.treemapSlice;
+        }
+        const treemap = d3.treemap()
+            .round(true)
+            .tile(tileStyle)
+            .size([this.state.visualizationWidth, 565])(root).leaves();
+
+        const colors = [
```

Comment:
We don't need to recreate this array with every `render()` cycle - let's move this near `propTypes` as it's own structure.